AnnotationEntry.setEntryNumber incorrectly returns ErrRSLEntryNotFound when RSL is empty

### What happened?

AnnotationEntry.setEntryNumber returns err on its final line instead of nil. When the RSL is empty, GetLatestEntry returns ErrRSLEntryNotFound, the number is correctly set to 1, but the error is then propagated to the caller, causing AnnotationEntry.Commit and CommitUsingSpecificKey to fail with ErrRSLEntryNotFound even though the operation logically succeeded.

All three entry types implement setEntryNumber with identical logic but AnnotationEntry alone returns err instead of nil:

Fix: Change the final line of AnnotationEntry.setEntryNumber from return err to return nil, consistent with ReferenceEntry and PropagationEntry.

gittuf revision: latest main as of June 16, 2026

### Relevant log output

```Shell
--- FAIL: TestAnnotationSetEntryNumberBug (0.07s)
    annotation_bug_test.go:16: setEntryNumber returned unexpected error on empty RSL: unable to find RSL entry
FAIL
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/internal/rsl/rsl.go
+++ b/internal/rsl/rsl.go
@@ -335,7 +335,7 @@ func (a *AnnotationEntry) setEntryNumber(repo *gitinterface.Repository) error {
 		}
 	}
 
-	return err
+	return nil
 }
 
 func (a *AnnotationEntry) createCommitMessage(includeNumber bool) (string, error) {

